packages/arb-token-bridge-ui/src/util/NetworkUtils.ts

```diff
@@ -0,0 +1,120 @@
+import ArbitrumNovaLogo from '../assets/ArbitrumNovaLogo.png'
+
+/* 
+  The list which will be available for network selection in navbar-dropdowns
+ */
+export const networkSelectionList = [
+  ChainId.ArbitrumOne,
+  ChainId.ArbitrumNova,
+  ChainId.Mainnet
+]
+
+/* 
+  Icons and colors attached with our networks available for selection
+*/
+export const networkStyleMap: {
+  [chainId: number]: { img: string; btnThemeClass: string }
+} = {
+  [ChainId.Mainnet]: {
+    img: EthereumLogo,
+    btnThemeClass: 'bg-blue-arbitrum'
+  },
+  [ChainId.ArbitrumOne]: {
+    img: ArbitrumOneLogo,
+    btnThemeClass: 'bg-blue-arbitrum'
+  },
+  [ChainId.ArbitrumNova]: {
+    img: ArbitrumNovaLogo,
+    btnThemeClass: 'bg-brick-dark'
+  }
+}
+
+/*
+  - A very basic variant of `changeNetwork` function present in App.tsx (which initializes after the app load)
+  - The original App-injected version requires a network connection already
+  - This lightweight function doesn't require a connected network - simply switches the metamask network
+  - But it doesn't support very rich business logic as well, like showing custom alerts if user is connected to `Arbitrum` or not
+*/
+
+export type SwitchChainProps = {
+  chainId: ChainId
+  provider: Web3Provider
+  onSuccess?: () => void
+  onError?: (err?: Error) => void
+  onSwitchChainNotSupported?: () => void
+}
+
+const noop = () => {}
+export async function switchChain({
+  chainId,
+  provider,
+  onSuccess = noop,
+  onError = noop,
+  onSwitchChainNotSupported
+}: SwitchChainProps) {
+  // first check if it is a valid network
+  let network
+  try {
+    network = await getL2Network(chainId)
+  } catch {
+    network = await getL1Network(chainId)
+  }
+  if (!network) return
+
+  // then check if the onSwitchChainNotSupported is passed, that means the switching is not supported
+  if (typeof onSwitchChainNotSupported === 'function') {
+    return onSwitchChainNotSupported()
+  }
+
+  // if all the above conditions are satisfied go ahead and switch the network
+  const hexChainId = hexValue(BigNumber.from(chainId))
+  const networkName = getNetworkName(chainId)
+
+  if (provider) {
+    console.log('Attempting to switch to chain', chainId)
+    try {
+      await provider.send('wallet_switchEthereumChain', [
+        {
+          chainId: hexChainId
+        }
+      ])
+
+      onSuccess?.()
+    } catch (err: any) {
+      if (err.code === 4902) {
+        console.log(`Network ${chainId} not yet added to wallet; adding now:`)
+        await provider.send('wallet_addEthereumChain', [
+          {
+            chainId: hexChainId,
+            chainName: networkName,
+            nativeCurrency: {
+              name: 'Ether',
+              symbol: 'ETH',
+              decimals: 18
+            },
+            rpcUrls: [rpcURLs[chainId]],
+            blockExplorerUrls: [getExplorerUrl(chainId)]
+          }
+        ])
+
+        onSuccess?.()
+      } else {
+        onError?.(err)
+        throw err
```

Comment:
We should either re-throw here or call `onError`, but not both. I'm for `onError`.